backend/ledgers.ts

```diff
@@ -12,94 +10,124 @@ interface Ledger {
+  horizon
+    .ledgers()
+    .cursor("now")
+    .limit(200)
+    .stream({
+      onmessage: async (ledger: LedgerRecord) => {
+        await updateCache([ledger], REDIS_LEDGER_KEY, REDIS_PAGING_TOKEN_KEY);
+      },
     });
 }
 
-function convertFields(ledger: LedgerSql): Ledger {
-  return {
-    // Remove year from date
-    date: ledger["date"].substring(5),
-    transaction_count: parseInt(ledger.transaction_count),
-    operation_count: parseInt(ledger.operation_count),
-  };
+export async function catchup(
+  ledgersKey: string,
+  pagingToken: string,
+  pagingTokenKey: string,
+  limit: number, // if 0, catchup until now
+) {
+  const horizon = new stellarSdk.Server("https://horizon.stellar.org");
+  let ledgers = [];
+  let total = 0;
+
+  while (true) {
+    let resp = await horizon.ledgers().cursor(pagingToken).limit(200).call();
+    ledgers = resp.records;
+    total += resp.records.length;
+    if (ledgers.length == 0 || (limit && total > limit)) {
+      break;
+    }
+
+    pagingToken = ledgers[ledgers.length - 1].paging_token;
+    await updateCache(ledgers, ledgersKey, pagingTokenKey);
+  }
 }
 
-// Wait for schema sync
-postgres.sequelize.addHook("afterBulkSync", () => {
-  setInterval(updateResults, 60 * 1000);
-  updateResults();
-
-  if (process.env.UPDATE_DATA == "true") {
-    // Stream ledgers - get last paging_token/
-    postgres.LedgerStats.findOne({ order: [["sequence", "DESC"]] }).then(
-      (lastLedger: postgres.LedgerStats) => {
-        let pagingToken;
-        if (!lastLedger) {
-          pagingToken = "now";
-        } else {
-          pagingToken = lastLedger.paging_token;
-        }
-
-        var horizon = new stellarSdk.Server("https://horizon.stellar.org");
-        horizon
-          .ledgers()
-          .cursor(pagingToken)
-          .limit(200)
-          .stream({
-            // TODO - use type any until https://github.com/stellar/js-stellar-sdk/issues/731 resolved
-            onmessage: (ledger: any) => {
-              let newLedger: any = pick(ledger, [
-                "sequence",
-                "closed_at",
-                "paging_token",
-                "operation_count",
-              ]);
-              newLedger["transaction_count"] =
-                ledger.successful_transaction_count +
-                ledger.failed_transaction_count;
-              postgres.LedgerStats.create(newLedger).then(() => {
-                console.log(
-                  "Added Ledger:" + ledger.sequence + " " + ledger.closed_at,
-                );
-              });
-            },
-          });
-      },
-    );
+export async function updateCache(
+  ledgers: Array<LedgerRecord>,
+  ledgersKey: string,
+  pagingTokenKey: string,
+) {
+  if (!ledgers.length) {
+    console.log("no ledgers to update");
+    return;
   }
-});
+  const json = (await redisClient.get(ledgersKey)) || "[]";
+  let cachedLedgers: Array<Ledger> = JSON.parse(json);
```

Comment:
Could be `const`

Reply:
https://github.com/stellar/dashboard/pull/209/commits/3668f052cf5b80466181349192ea2a67ed6e1e36